design/fast_finality.md

```diff
@@ -0,0 +1,276 @@
+    of transactions (e.g. by running a single transaction, then `eth_call`ing to see how it has affected a smart 
+    contract's state)
+* Cost
+  * L1 transaction costs can be driven lower at the expense of extending the hard-finality window
+* User/dev experience
+  * The responses to RPC calls reflect the soft-finalised transactions, and not just the hard-finalised transactions
+* Resilience
+  * Failover does not require a governance action
+  * During failover, it is acceptable to break the one-second soft-finality guarantee 
+  * During failover, it is acceptable to drop transactions that have not been soft-finalised
+
+## Assumptions
+
+* There is a single aggregator that is also the genesis node. This node is known as the *sequencer*
+* The enclave has a start-up delay that exceeds the production time for light batches (see below)
+
+## Constraints
+
+* It must be possible to only publish rollups to the L1 every `x` blocks, where `x` >> 1
+* It must be possible to produce light batches (see below) at a higher frequency than L1 blocks
+
+## Design
+
+### Sequencer identity
+
+The sequencer's identity is given in the management contract on the L1 as a set of enclave attestations. This allows 
+other nodes to verify that the light batches and rollups are created by the sequencer. We use a set instead of a 
+single attestation to allow faster failover in the case of a sequencer node crashing (see the section `Resilience`, 
+below).
+
+Each attestation matches one of the sequencer's enclaves, and contains the hash of that enclave's key. Since the
+attestations are unique per machine, the enclaves cannot be impersonated. The foundation admin will then whitelist
+these attestations. If one of the sequencer's enclaves goes down irrecoverably, a replacement attestation is added and
+whitelisted, following the same process.
+
+### Node start-up
+
+At start-up, the host checks if one of the following applies:
+
+* They are the genesis node and are an aggregator
+* They are *not* the genesis node and are *not* an aggregator
+
+If neither of these conditions is met, the host shuts down.
+
+### Production of light batches
+
+A light batch is produced on the required cadence to meet the network's soft-finality window of one second. Only the 
+sequencer produces light batches.
+
+To produce a light batch, the sequencer's host feeds a set of transactions to the enclave. The enclave responds by 
+creating a signed and encrypted *light batch*. This light batch is formally identical to the rollup of the final 
+design, including a list of the provided transactions and a header including information on the current light batch and 
+the hash of the "parent" light batch.
+
+The sequencer's host immediately distributes the light batch to all other nodes, who gossip it onwards to other nodes
+(ensuring the sequencer cannot restrict the distribution of light batches to specific nodes). These light batches are 
```

Comment:
How does this ensure the sequencer cannot restrict the distribution? Honest validators will of course notice that they have not received a Light Batch in good time so I guess that's what keeps the central sequencer honest.

Reply:
I extended it to "(ensuring the sequencer cannot restrict the distribution of light batches to specific nodes, provided one of the nodes who received the light batch is honest)" - the idea being that if one recipient is honest, they'll gossip it onwards.